internal-api/src/main/java/datadog/trace/bootstrap/config/provider/StableConfigSource.java

```diff
@@ -0,0 +1,95 @@
+package datadog.trace.bootstrap.config.provider;
+
+import static datadog.trace.util.Strings.propertyNameToEnvironmentVariableName;
+
+import datadog.trace.api.ConfigOrigin;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class StableConfigSource extends ConfigProvider.Source {
+  private static final Logger log = LoggerFactory.getLogger(StableConfigSource.class);
+
+  public static final String USER_STABLE_CONFIG_PATH =
+      "/etc/datadog-agent/application_monitoring.yaml";
+  public static final String MANAGED_STABLE_CONFIG_PATH =
+      "/etc/datadog-agent/managed/datadog-agent/stable/application_monitoring.yaml";
+  public static final StableConfigSource USER =
+      new StableConfigSource(USER_STABLE_CONFIG_PATH, ConfigOrigin.USER_STABLE_CONFIG);
+  public static final StableConfigSource MANAGED =
+      new StableConfigSource(
+          StableConfigSource.MANAGED_STABLE_CONFIG_PATH, ConfigOrigin.MANAGED_STABLE_CONFIG);
+
+  private final ConfigOrigin fileOrigin;
+
+  private final StableConfig config;
+
+  StableConfigSource(String file, ConfigOrigin origin) {
+    this.fileOrigin = origin;
+    StableConfig cfg;
+    try {
+      cfg = StableConfigParser.parse(file);
+    } catch (Throwable e) {
+      log.debug("Stable configuration file not readable at specified path: {}", file);
+      cfg = new StableConfig();
```

Comment:
```suggestion
      cfg = StableConfig.EMPTY;
```

you can then remove the default constructor from `StableConfig`